starter/src/main/java/io/javaoperatorsdk/operator/springboot/starter/OperatorAutoConfiguration.java

```diff
@@ -57,14 +58,24 @@ public KubernetesClient kubernetesClient(Optional<HttpClient.Factory> httpClient
                 .build());
   }
 
-
-
   @Bean
   @ConditionalOnMissingBean(ResourceClassResolver.class)
   public ResourceClassResolver resourceClassResolver() {
     return new DefaultResourceClassResolver();
   }
 
+  @Bean
+  public CrdUploader crdUploader(KubernetesClient client, List<CrdTransformer> transformers) {
```

Comment:
So what I mean with conditional property is to use `@ConditionalOnProperty` here, and create the bean and apply the property has the correct value.
And make sure it happens before the operator bean created, just by adding an optional param for CRDUploader for Operator.

Reply:
I think it's a bit hacky, I'd rather have this as functional interface and have a on missing bean that would just log that it did nothing or event just no log , wdyt?
Like
``` 
@ConditionalOnProperty(enable)
my actual bean(){}

@ConditionalOnProperty(enable) or just @ConditionalOnMissingBean
() -> log.debug("Not applying bla bla")

```

Reply:
sounds great!